File: programmers/Lv_2/Kimyongho/짝지어제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        int answer = 0;
        
        Stack<Character> stack = new Stack<Character>();
        stack.push(s.charAt(0));
		for(int i = 1; i < s.length(); i++) {
			char c = s.charAt(i);
			if(stack.empty()) {
				stack.push(c);
			}
			else {
				if(!(stack.peek() == s.charAt(i))) {
					stack.push(c);
				}
				else {
					stack.pop();
				}
			}
		}
		if(stack.empty()) {
			return 1;
		}
        return answer;
    }
}

File: programmers/Lv_2/HwangJunghoon/짝지어 제거하기.py
def solution(s):
    # stack 사용 : lst로 담음
    lst=[]
    for i in s:
        lst.append(i)
        if len(lst)>1:
            if lst[-2]==i:
                lst.pop()
                lst.pop()
    if len(lst)==0:
        return 1
    else:
        return 0